inventory/sample/group_vars/all.yml

```diff
@@ -22,16 +22,18 @@ k3s_token: "some-SUPER-DEDEUPER-secret-password"
 # it for each of your hosts, though.
 k3s_node_ip: '{{ ansible_facts[flannel_iface]["ipv4"]["address"] }}'
 
-k3s_single_node: "{{ 'true' if groups['k3s_cluster'] | length == 1 else 'false' }}"
+k3s_node_exists: "{{ 'true' if groups['node'] | default([]) | length >= 1 else 'false' }}"
 
 # these arguments are recommended for servers as well as agents:
 extra_args: >-
   --flannel-iface={{ flannel_iface }}
   --node-ip={{ k3s_node_ip }}
 
-# change these to your liking, the only required one is --disable servicelb
+# change these to your liking, the only required are: --disable servicelb, --tls-san {{ apiserver_endpoint }}
 extra_server_args: >-
   {{ extra_args }}
+  {{ '--node-taint node-role.kubernetes.io/master=true:NoSchedule' if k3s_node_exists | bool else '' }}
+  --tls-san {{ apiserver_endpoint }}
```

Comment:
@ioagel You are writing:

> The new addition in extra_server_args: --tls-san {{ apiserver_endpoint }} is mandatory, because the ip of our virtual ip needs to be in the SANs of the api server certificate, otherwise we cannot access our cluster.

I suppose you did experience this error. I'm curious what went wrong there as cert validation is active during molecule testing (which accesses the API server from the outside via the VIP). In my local cluster, the VIP is presented as a SAN also - without that argument. So I'm wondering whether k3s has an automatism for determining possible subject names that the playbook breaks in some case... Do you have any insight here?

Reply:
Yes, I experienced the error and in every k3s cluster installation (using usually k3sup) I have done, without setting the SANs properly. Usually in front of the clusters I used a haproxy external instance, which without setting it’s ip in the TLS SAN, I could not access the cluster.
Now… I am still puzzled that you do not have the same problem…